src/gui/sdlmain.cpp

```diff
@@ -1509,23 +1509,32 @@ static GLuint BuildShader(GLenum type, const std::string& source)
 	glCompileShader(shader);
 
 	// Check the compile status
-	glGetShaderiv(shader, GL_COMPILE_STATUS, &compiled);
+	glGetShaderiv(shader, GL_COMPILE_STATUS, &is_shader_compiled);
 
-	if (!compiled) {
-		GLint info_len = 0;
-		glGetShaderiv(shader, GL_INFO_LOG_LENGTH, &info_len);
+	GLint info_len = 0;
+	glGetShaderiv(shader, GL_INFO_LOG_LENGTH, &info_len);
 
-		if (info_len > 1) {
-			std::vector<GLchar> info_log(info_len);
-			glGetShaderInfoLog(shader, info_len, nullptr, info_log.data());
-			LOG_ERR("Error compiling shader: %s", info_log.data());
+	// The info log might contain warnings and info messages even if the
+	// compilation was successful, so we'll always log it if it's non-empty.
+	if (info_len > 1) {
+		std::vector<GLchar> info_log(info_len);
+		glGetShaderInfoLog(shader, info_len, nullptr, info_log.data());
+
+		if (is_shader_compiled) {
+			LOG_WARNING("SDL:OPENGL: Shader info log: %s",
+			            info_log.data());
+		} else {
+			LOG_ERR("SDL:OPENGL: Error compiling shader: %s",
+			        info_log.data());
```

Comment:
I think these should just be prefixed with `OPENGL:` rather than `SDL:OPENGL:`.  I see in my logs where we use one or the other of these prefixes but never both in the same log line.

Reply:
There are few more such cases for MIDI and VGA logging but just using `OPENGL` makes sense, so I'll do as you suggested.

Behold my crappy regex skills! 😆 

<img width="1289" alt="image" src="https://github.com/dosbox-staging/dosbox-staging/assets/698770/0c487b5d-41c2-441b-b4aa-a9c95606f4e8">

Reply:
Yeah, it makes a bit more sense for MIDI and VGA I think.  A user might not know what ALSA is and I don't even know what those VGA sub-systems are.

`sdlmain.cpp` logging looks inconsistent though and it appears there are a few other places that use both prefixes.  I would prefer drawing the line at which function you last called when you got the error.  Here's an OpenGL function that's returned an error.  Although I guess you could make the argument that it's in our internal SDL "module".  I'll let you make the final judgement call on what you want to do.

Reply:
Agreed @weirddan455. I made a judgement call on when to use the `OPENGL:` and `SDL:` prefixes, to me it makes sense but feel free to disagree. Please have a look at the new logging improvement commits.

Reply:
The line is kinda blurry on some of those where it's an SDL function and error returned by `SDL_GetError()` but it's clearly related to OpenGL.  I think these new logging changes are positive though.  It makes it more consistent.

Reply:
Yeah I said, judgement call, but my reasoning is SDL  acts as a very thin wrapper to OpenGL in those cases, and those are still legitimate OpenGL errors that SDL just forwards (e.g. not being able to request an sRGB framebuffer; that's an OpenGL issue, SDL just acts as a "messenger").